server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -306,6 +289,25 @@ public DocumentMapper merge(Mapping mapping, MergeReason reason) {
         return new DocumentMapper(mapperService, merged);
     }
 
+    public void validate(IndexSettings settings, boolean checkLimits) {
+        this.mapping.validate(this.fieldMappers);
+        if (settings.getIndexMetadata().isRoutingPartitionedIndex()) {
+            if (routingFieldMapper().required() == false) {
+                throw new IllegalArgumentException("mapping type [" + type() + "] must have routing "
+                    + "required for partitioned index [" + settings.getIndex().getName() + "]");
+            }
+        }
+        if (settings.getIndexSortConfig().hasIndexSort() && hasNestedObjects()) {
+            throw new IllegalArgumentException("cannot have nested fields when index sort is activated");
+        }
+        if (checkLimits) {
+            this.fieldMappers.checkFieldLimit(settings.getMappingTotalFieldsLimit());
```

Comment:
Small comment, we could expose all these checks as a single method `DocumentFieldMappers#checkLimits(IndexSettings settings)`.

Reply:
++